### PR TITLE
[Clang][Sema] placement new initializes typedef array with correct size

### PR DESCRIPTION
**Pull request by mahtohappy**
Build Failure Fix
Fixes build failures due to #83124

### CODE REVIEW

File: clang/test/SemaCXX/PR41441.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
-// RUN: %clang -S -fno-discard-value-names -emit-llvm -o - %s | FileCheck %s
-// Issue no: 41441
+// RUN: %clang --target=x86_64-pc-linux -S -fno-discard-value-names -emit-llvm -o - %s | FileCheck %s
+
 #include <new>
```
Comment (erichkeane) [inline clang/test/SemaCXX/PR41441.cpp:3]: This #include is also possible to cause breakages, you should just put hte important declarations in text.

**Review by erichkeane — APPROVED** [review]
This should be fine to just merge, don't need to wait for pre-commit CI.

File: clang/test/SemaCXX/PR41441.cpp (hunk 1 above)
Comment (dyung) [reply clang/test/SemaCXX/PR41441.cpp:3]: It does:

https://lab.llvm.org/buildbot/#/builders/280/builds/1989
```
/Users/buildbot/buildbot-root/aarch64-darwin/llvm-project/clang/test/SemaCXX/PR41441.cpp:3:10: fatal error: 'new' file not found
    3 | #include <new>
      |          ^~~~~
1 error generated.
```

Comment (erichkeane) [reply clang/test/SemaCXX/PR41441.cpp:3]: Blarg, the buildbots i checked on were mangling differences.  I'll fix this locally as I suggested and get a fix merged ASAP.

Comment (erichkeane) [reply clang/test/SemaCXX/PR41441.cpp:3]: Should be good as of 0a789ea8a

Comment (nico) [reply clang/test/SemaCXX/PR41441.cpp:3]: Thanks!

**Pull request merged**